File: library/src/androidTest/mock/com/mobilesolutionworks/android/app/test/basic/TestWorksActivity.java
package com.mobilesolutionworks.android.app.test.basic;

import android.os.Bundle;

import com.mobilesolutionworks.android.app.WorksActivity;
import com.mobilesolutionworks.android.app.controller.WorksControllerManager;
import com.mobilesolutionworks.android.app.test.GetController;
import com.mobilesolutionworks.android.app.test.R;

/**
 * Created by yunarta on 12/3/17.
 */

public class TestWorksActivity extends WorksActivity implements GetController<TestWorksController> {

    private TestWorksController mController;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.fragment_host_activity);

        mController = getControllerManager().initController(0, null, new WorksControllerManager.ControllerCallbacks<TestWorksController>() {
            @Override
            public TestWorksController onCreateController(int id, Bundle args) {
                return new TestWorksController();
            }
        });

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .replace(com.mobilesolutionworks.android.app.test.R.id.fragment_container, new TestWorksFragment(), "root")
                    .commitNow();
        }

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mController = null;
    }

    public TestWorksController getController() {
        return mController;
    }
}

File: test-app/src/main/kotlin/com/mobilesolutionworks/android/controller/test/basic/TestWorksActivity.kt
package com.mobilesolutionworks.android.controller.test.basic

import android.os.Bundle
import com.mobilesolutionworks.android.app.WorksActivity
import com.mobilesolutionworks.android.app.controller.CreateCallback2
import com.mobilesolutionworks.android.controller.test.GetController
import com.mobilesolutionworks.android.controller.test.R

/**
 * Created by yunarta on 12/3/17.
 */
open class TestWorksActivity : WorksActivity(), GetController<TestWorksController> {

    override var controller: TestWorksController? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.fragment_host_activity)

        controller = controllerManager.initController(0, null, CreateCallback2 { TestWorksController(it) })

        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                    .replace(R.id.fragment_container, TestWorksFragment(), "root")
                    .commitNow()
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        controller = null
    }
}
